gslib/tests/test_iam.py

```diff
@@ -1555,3 +1580,525 @@ def test_shim_translates_iam_set_handles_empty_etag(self):
+              'alpha',
+              'storage',
+              'objects',
+              'set-iam-policy',
+              'gs://b/dir/:',
+              '-',
+          ],
+                             stdin=mock.ANY),
+          self._get_run_call([
+              os.path.join('fake_dir', 'bin', 'gcloud'), 'alpha', 'storage',
+              'objects', 'get-iam-policy', 'gs://b/dir2/o', '--format=json'
+          ]),
+          self._get_run_call([
+              os.path.join('fake_dir', 'bin', 'gcloud'),
+              'alpha',
+              'storage',
+              'objects',
+              'set-iam-policy',
+              'gs://b/dir2/o',
+              '-',
+          ],
+                             stdin=mock.ANY)
+      ])
+
+  @mock.patch.object(iam.IamCommand, 'RunCommand', new=mock.Mock())
+  @mock.patch.object(subprocess, 'run', autospec=True)
+  def test_iam_ch_raises_ls_error(self, mock_run):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True')]):
+      ls_process = subprocess.CompletedProcess(args=[],
+                                               returncode=1,
+                                               stderr='An error.')
+      mock_run.side_effect = [ls_process]
+
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        with self.assertRaisesRegex(CommandException, 'An error.'):
+          self.RunCommand(
+              'iam', ['ch', 'allAuthenticatedUsers:modified-role', 'gs://b/o'])
+        self.assertEqual(mock_run.call_count, 1)
+
+  @mock.patch.object(iam.IamCommand, 'RunCommand', new=mock.Mock())
+  @mock.patch.object(subprocess, 'run', autospec=True)
+  def test_iam_ch_raises_get_error(self, mock_run):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True')]):
+      ls_process = subprocess.CompletedProcess(args=[],
+                                               returncode=0,
+                                               stdout=json.dumps([{
+                                                   'url': 'gs://b/o',
+                                                   'type': 'cloud_object'
+                                               }]))
+      get_process = subprocess.CompletedProcess(args=[],
+                                                returncode=1,
+                                                stderr='An error.')
+      mock_run.side_effect = [ls_process, get_process]
+
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        with self.assertRaisesRegex(CommandException, 'An error.'):
+          self.RunCommand(
+              'iam', ['ch', 'allAuthenticatedUsers:modified-role', 'gs://b/o'])
+        self.assertEqual(mock_run.call_count, 2)
+
+  @mock.patch.object(iam.IamCommand, 'RunCommand', new=mock.Mock())
```

Comment:
Why does `RunCommand` execute if the shim is active?

Reply:
I've removed those mocks.